File: Python/QuickSort.py

# Partition Method Whose is using into the Quick Method,
def Partition(DataStore, low, high):
    pivot = DataStore[high]
    i = low-1

    for x in range(low,high):
        if(DataStore[x]<=pivot):
            i = i+1
            temp = DataStore[i]
            DataStore[i] = DataStore[x]
            DataStore[x] = temp
        
    
    temp = DataStore[i+1]
    DataStore[i+1] = DataStore[high]
    DataStore[high] = temp

    return (i+1)


# Quick Sort Method, 
def Quick(DataStore, low, high):
    if(low<high):
        p = Partition(DataStore,low,high)
        Quick(DataStore,low,p-1)
        Quick(DataStore,p+1,high)


#Declear the DataStore & here already set the re-arrange data,
#In the size variable, declear the length of the DataStore,
DataStore = [24,9,29,14,19,27]
size = len(DataStore)
Quick(DataStore,0,size-1)
print(DataStore)

File: Exceptional_Directory/QuickSort.cpp
//Quick Sort, 

#include<iostream>
using namespace std;

int Partition(int DataStore[],int low,int high){
    int pivot = DataStore[high];
    int i = low-1;
    int temp = 0;
    for(int x=low; low<high; x++){
        if(pivot>=DataStore[x]){
            i++;
            temp = DataStore[i];
            DataStore[i] = DataStore[x];
            DataStore[x] = temp;
        }
    }
    temp = DataStore[i+1];
    DataStore[i+1] = DataStore[high];
    DataStore[high] = temp;

    return (i+1);
    
}


void QuickSort(int DataStore[], int low,int high){
    if(low<high){
        int p = Partition(DataStore,low,high);
        QuickSort(DataStore,low,p-1);
        QuickSort(DataStore,p+1,high);
    }
}




//print DataStore,
void PrintDataStore(int DataStore[], int size){
    cout<<"Show Data Store : ";
    for(int a=0; a<size; a++){
        cout<<DataStore[a]<<" ";
    }
    cout<<endl;
}


int main(){

int DataStore[] = {24,9,29,14,19,27};
int size = sizeof(DataStore)/sizeof(DataStore[0]);
int low = 0;
int high = size-1;
QuickSort(DataStore,low,high);
PrintDataStore(DataStore,size);

    return 0;
}